label_studio_ml/examples/yolo_sam2_tracker/control_models/video_rectangle_with_yolo_sam2_tracker.py

```diff
@@ -0,0 +1,1068 @@
+        )
+
+        return self.sam2_predictor
+
+    def convert_mask_to_bbox(self, mask: Mask):
+        """
+        Function to convert a mask to a bounding box.
+        Used from Label Studio ML examples.
+        :param mask:
+        :return:
+        """
+        # Decode the mask
+        mask_np = coco_mask.decode(mask.encoded)
+
+        # squeeze
+        logger.debug(f"Mask shape: {mask_np.shape}")
+
+        y_indices, x_indices = np.where(mask_np == 1)
+        if len(x_indices) == 0 or len(y_indices) == 0:
+            return None
+
+        # Find the min and max indices
+        xmin, xmax = np.min(x_indices), np.max(x_indices)
+        ymin, ymax = np.min(y_indices), np.max(y_indices)
+
+        # Get mask dimensions
+        height, width = mask_np.shape
+
+        # Calculate bounding box dimensions
+        box_width = xmax - xmin + 1
+        box_height = ymax - ymin + 1
+
+        # Normalize and scale to percentage
+        x_pct = (xmin / width) * 100
+        y_pct = (ymin / height) * 100
+        width_pct = (box_width / width) * 100
+        height_pct = (box_height / height) * 100
+
+        return {
+            "x": round(x_pct, 2),
+            "y": round(y_pct, 2),
+            "width": round(width_pct, 2),
+            "height": round(height_pct, 2)
+        }
+
+    def split_frames(self, video_path, temp_dir, video_fps, start_frame=0, end_frame=100):
+        """
+        Extracts and saves frames from a video file within the specified range.
+
+        This method processes a video file, extracts frames between the provided
+        start and end frame indexes, and stores the extracted images in the specified
+        temporary directory. Each frame is saved as a `.jpg` file, and the method
+        yields the file path and the corresponding frame data for further processing.
+
+        :param video_path: Path to the input video file.
+        :param temp_dir: Directory to store the extracted frame images temporarily.
+        :param start_frame: Index of the first frame to extract (inclusive).
+        :param end_frame: Index of the last frame to extract (exclusive).
+        :return: Yields a tuple containing the file path and raw frame data for each extracted frame.
+        """
+        logger.debug(f'Opening video file: {video_path}')
+
+        # Get video properties using the static method
+        frame_count, duration = self.get_video_fps_duration(video_path, fps=video_fps)
+
+        # Open the video using OpenCV
+        video = cv2.VideoCapture(video_path)
+
+        if not video.isOpened():
+            raise ValueError(f"Could not open video file: {video_path}")
+
+        logger.debug(f'Video duration: {duration} seconds, {frame_count} frames, {video_fps} fps')
+
+        frame_count_current = 0
+        while frame_count_current < frame_count:
+            # Calculate the timestamp in seconds for the current frame
+            timestamp = frame_count_current / video_fps
```

Comment:
I'm not able to read the video based on the defined fps in the code_template. Currently runs for 5fps. I want to make this fps configurable through frontend.